compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt

```diff
@@ -13,12 +13,112 @@ import org.jetbrains.kotlin.psi.psiUtil.getValueParameters
 import org.jetbrains.kotlin.psi.psiUtil.modalityModifierType
 import org.jetbrains.kotlin.psi.psiUtil.visibilityModifierType
 
+/**
+ * [classOrObject] is a function that intercepts all [KtClass] elements that [match]
+ * allowing a [Transform] to change the intercepted AST tree before compilation.
+ *
+ * An extension function of [Meta] and inheriting from [ExtensionPhase], [classOrObject] was designed to to feed in
+ * virtually any kind of [KtClass] predicate, followed by mapping a function that takes the desired [Scope] of our
```

Comment:
```suggestion
 * virtually any kind of [KtClass] predicate, followed by a mapping function that takes the desired [Scope] of our
```
Is this accurate as is? I'm not 100% sure about this one since I don't have full understanding of the codebase yet, but from my point of view it feels like it's a mapping function what follows, instead of "mapping a function". It might also be because of my not very high English level :D

Reply:
I think you make a really good point - at the very least, this suggests we need to clarify what we mean.

I want to say yes though - I think I talk about it more below with this section of code:

``` ```
 *  val Meta.lenses: Plugin
 *    get() =
 *      "lenses" {
 *        meta {
 *          classOrObject(::isProductOrType) { c -> ... }
 *        }
 *      }
 * ```

You can see the pattern where you match and you do something with the resulting match. 
Since the second parameter of `classOrObject` is 
```map: ClassScope.(KtClass) -> Transform<KtClass>```

We always know we're applying a function to the resulting predicate coming out of the analysis phase

Reply:
Yeah I'd put this as a "mapping function"